html/pfappserver/root/static.alt/src/components/pfFormPassword.vue

```diff
@@ -0,0 +1,194 @@
+    class="pf-form-password" :class="{ 'mb-0': !columnLabel, 'is-focus': focus }">
+    <b-input-group class="input-group-password">
+      <b-form-input
+        v-model="inputValue"
+        v-bind="$attrs"
+        ref="input"
+        :type="type"
+        :state="isValid()"
+        @input.native="validate()"
+        @keyup.native="resetTest($event)"
+        @change.native="onChange($event)"
+        @focus.native="focus = true"
+        @blur.native="focus = false"
+      >
+      </b-form-input>
+      <b-input-group-append>
+        <b-button-group v-if="test" rel="testResultGroup">
+          <b-button v-if="testResult !== null" variant="light" disabled tabindex="-1">
+            <span class="mr-1" :class="{ 'text-danger': !testResult, 'text-success': testResult }">{{ testMessage }}</span>
+          </b-button>
+        </b-button-group>
+        <b-button-group rel="prefixButtonGroup">
+          <b-button v-if="test" class="input-group-text" @click="runTest()" :disabled="isLoading || !this.value" tabindex="-1">
+            {{ $t('Test') }}
+            <icon v-show="isTesting" name="circle-notch" spin class="ml-2 mr-1"></icon>
+            <icon v-if="testResult !== null && testResult" name="check" class="ml-2 mr-1 text-success"></icon>
+            <icon v-if="testResult !== null && !testResult" name="times" class="ml-2 mr-1 text-danger"></icon>
+          </b-button>
+          <b-button class="input-group-text" @click="click()" @mouseover="over()" @mousemove="over()" @mouseout="out()" :disabled="!this.value && this.type === 'password'" :pressed="visible" tabindex="-1"><icon name="eye"></icon></b-button>
+        </b-button-group>
+      </b-input-group-append>
+    </b-input-group>
+    <b-form-text v-if="text" v-t="text"></b-form-text>
+  </b-form-group>
+</template>
+
+<script>
+import pfMixinValidation from '@/components/pfMixinValidation'
+
+export default {
+  name: 'pf-form-password',
+  mixins: [
+    pfMixinValidation
+  ],
+  props: {
+    value: {
+      default: null
+    },
+    columnLabel: {
+      type: String
+    },
+    labelCols: {
+      type: Number,
+      default: 3
+    },
+    text: {
+      type: String,
+      default: null
+    },
+    type: {
+      type: String,
+      default: 'password'
+    },
+    test: {
+      type: Function,
+      default: null
+    },
+    isLoading: {
+      type: Boolean,
+      default: false
+    }
+  },
+  data () {
+    return {
+      visible: false,
+      focus: false,
+      testResult: null,
+      testMessage: null,
+      isTesting: false
+    }
+  },
+  computed: {
+    inputValue: {
+      get () {
+        return this.value
+      },
+      set (newValue) {
+        this.$emit('input', newValue)
+      }
+    }
+  },
+  methods: {
+    over () {
+      this.type = 'text'
+    },
+    out () {
+      this.type = (this.visible) ? 'text' : 'password'
+    },
+    click () {
+      this.visible = !this.visible
```

Comment:
Are `over` and `out` called when using a touchscreen? If not, this line will have no effect.